### ISSUE TITLE
Checklist for v0.2 tag

### ISSUE THREAD

**Issue by xylar** [MPAS-Dev/MPAS-Analysis#154]
### To do list for MPAS-Analysis v. 0.2 and subsequent ACME analysis update:

- [x] @pwolfram tests if #151 works with the branch of xarray he's working to have in the next version
- [x] @milenaveneziani or @pwolfram merges #151 into `MPAS-Analysis/develop`
- [x] @milenaveneziani merges #153 
- [x] @milenaveneziani merges #157 
- [x] @pwolfram merges #163 (which will be helpful for testing runs on LANL machines)
- [x] @xylar merges #166 
- [x] @xylar creates a test branch `MPAS-Analysis/master_v0.2` where he merges `MPAS-Analysis/develop` into `MPAS-Analysis/master`
- [x] @xylar updates the `MPAS-Analysis` submodule in [PreAndPostProcessingScripts/xylar/update_MPAS_Analysis_v0.2](https://github.com/ACME-Climate/PreAndPostProcessingScripts/tree/xylar/update_MPAS_Analysis_v0.2) to point to the test branch
- [x] Other updates in [PreAndPostProcessingScripts/xylar/update_MPAS_Analysis_v0.2](https://github.com/ACME-Climate/PreAndPostProcessingScripts/tree/xylar/update_MPAS_Analysis_v0.2) for `v0.2` support:
    - [x] update variables in run*.csh scripts (@xylar)
    - [x] update variables in `python/setup_ocnice_config.py` (@xylar)
    - [x] update html generation (@milenaveneziani)
- [x] Updates to observations, mapping and mask files:
    - [x] Add mapping files for EC60to30v3_to_0.5x0.5degree on all machines (@milenaveneziani)
      - [x] edison
      - [x] lanl
      - [x] aims
      - [x] rhea 
    - [x] Add MOC region/transect files for all supported grids (@xylar) and move them to all machines
      - [x] oEC60to30v1, oEC60to30v3, oEC60to30wLIv1, oQU240v1, oRRS18to6v1
      - [x] edison
      - [x] lanl
      - [x] aims (@milenaveneziani)
      - [x] rhea (@milenaveneziani)
- [x] Preliminary testing on:
    - [x] edison (@xylar)
      - [x] beta0
      - [x] beta1
    - [x] lanl (@xylar)
- [ ] Coordinate with Workflow team to update anaconda on various machines:
    - [x] establish common list of anaconda 2.7 modules
    - [x] establish list of machines
    - [ ] establish maintainers for anaconda on each machine
    - [ ] get current package list of each machine
    - [ ] update edison (@xylar)
    - [ ] update cori (@xylar)
    - [ ] update lanl (@xylar)
    - [x] update acme1 (@williams13)
    - [ ] update aims (@milenaveneziani)
    - [ ] update rhea / titan (@pwolfram) 
- [ ] Test [PreAndPostProcessingScripts/xylar/update_MPAS_Analysis_v0.2](https://github.com/ACME-Climate/PreAndPostProcessingScripts/tree/xylar/update_MPAS_Analysis_v0.2)  on:
    - [x] edison (@xylar)
    - [ ] cori (@xylar if there are any model results on Cori to test...) 
    - [x] lanl (@xylar)
    - [ ] aims (@milenaveneziani)
    - [ ] rhea (@milenaveneziani) 
- [ ] @xylar pushes our test branch to `MPAS-Analysis/master` and creates a `v0.2` tag.  (I don't think this will change the commit for the submodule in [PreAndPostProcessingScripts/xylar/update_MPAS_Analysis_v0.2](https://github.com/ACME-Climate/PreAndPostProcessingScripts/tree/xylar/update_MPAS_Analysis_v0.2), so no update should be needed on the ACME side, but @xylar will make sure.
- [ ] @xylar makes a PR for [PreAndPostProcessingScripts/xylar/update_MPAS_Analysis_v0.2](https://github.com/ACME-Climate/PreAndPostProcessingScripts/tree/xylar/update_MPAS_Analysis_v0.2).  (@kevans32 and @salilmahajan will review; @kevans32 will merge)

**Comment by milenaveneziani** [comment]
@pwolfram: do I have permission to update conda on rhea?

**Comment by pwolfram** [comment]
@milenaveneziani, I don't think so.  I'll update conda once xarray is pushed to the minor version number by @shoyer.

**Comment by xylar** [comment]
@milenaveneziani, if you can point me to a run with the EC60to30v3 grid, I can make mapping files and region masks for that.  Then, I should be able to replace the old region mask for the v1 grid (with the less accurate southern transect) with an updated version at the same time I put in a region mask file and a mapping file for the v3 grid.

**Comment by xylar** [comment]
@milenaveneziani, there's an issue with the `index.html` not pointing to the right file names if the climatology (or, in many cases, time series) did not include all of the requested years.  I don't know how the code for creating the html file can know what the actual range of available output was to correct for this problem.  Perhaps we say for now that the html doesn't work unless you make sure to set the start and end years to be within the range of the actual results?

**Comment by milenaveneziani** [comment]
that's a good point @xylar.. There is another csh script that makes a case.tmp file, storing env variables in it, including the begin_yr and end_yr values that are set in run_*.csh. So, yes, those will be the intended years, so the real ones.
One option could be to test, in run_*.csh, that the choices of begin_yr,end_yr are appropriate for the particular data set in question, and then issue an error if they are not. In fact, I don't know if the atmosphere diagnostics support an out-of-range case at all, so that would be a good idea in general for the ACME diags.

**Comment by xylar** [comment]
@milenaveneziani, feel free to add such a test and push it to my ACME analysis branch.

**Comment by xylar** [comment]
To test on NERSC with the updated anaconda configuration, please modify the appropriate line of your custom `run_EDISON.csh` to be:
```
module load python/anaconda-2.7-climate_test
```
Essentially simultaneously with the merge of https://github.com/ACME-Climate/PreAndPostProcessingScripts/pull/29, I will update the shared anaconda environment to match this one.

**Comment by milenaveneziani** [comment]
@xylar: I am thinking about adding a simple test for this, considering that the cshell scripts will be changing/disappearing in the near future anyway. I would parse in the year value that is in a rpointer file: that would be part of the restart date for the next restart cycle, thus indicating that data is currently available for years<rpointer_date. That of course will not mean that *all* files will be available between year=1 and year=rpointer_date-1, but it seems like the easiest way to do this in a temporary manner.
What do you think?

**Comment by xylar** [comment]
@milenaveneziani, So if I understand right, you would take the values read in but change the end year if it's beyond the date in the restart pointer?  That sounds reasonable to me.  Eventually, we'll probably want to just parse it out of the file names or the output config files themselves but that's much easier to do in python.

**Comment by milenaveneziani** [comment]
If end_yr_climo or end_yr_ts are greater than the year in the rpointer file, I was actually just thinking to issue a warning and exit the cshell script all together.
This is because I am not sure the atm diagnostics can even handle the case when end_yr is greater than the max available year.
We can improve when everything is in python, of course.

**Comment by xylar** [comment]
Ah, right.  You said that was the plan and I just forgot.

**Comment by xylar** [comment]
@pwolfram, do you have an estimate on how soon xarray 0.9.2 will be released?  Based on my experience last week, there seems to be urgent need of us updating the ACME diagnostics and I think everything is in place for v0.2.  Assuming xarray 0.9.2 won't be released by early this coming week, it might be best to go ahead with v0.2 and updating to xarray 0.9.1.  Then, we can update again when 0.9.2 is released and also address any problems that might arise (with our own v0.2.1 update if need be).

**Comment by pwolfram** [comment]
@xylar I will follow up with @shoyer and get back to you.

**Comment by pwolfram** [comment]
@xylar, it sounds like a release within the next day is reasonable so "early this coming week" should be met.  Will that work?

**Comment by xylar** [comment]
Yes, I think that's fine.  It will be easier for us if we don't have to update xarray twice.

**Comment by pwolfram** [comment]
@xylar, @shoyer graciously released xarray to 0.9.2 last night so we should be good.

**Comment by pwolfram** [comment]
We can update xarray on each platform now for testing of 0.2

**Comment by xylar** [comment]
That's great!  I just updated on my laptop and will update Edison now.  Wolf soon.

**Comment by xylar** [comment]
@pwolfram, did you update the xarray version used by everyone on Titan/Rhea? Or just a test install that we will use to update the conda everyone else uses later on?  I ask because the current ACME analysis won't work with xarray > 0.8.2 because of a bug in SSS climatology calculations.

**Comment by pwolfram** [comment]
I just updated it for everyone.  I'll downgrade it now-- thank you very much for catching this for me.

**Comment by xylar** [comment]
So my plan was to create a second anaconda install (or an environment if you prefer, but I find it cleaner with a completely separate install) for testing.  This is what I have done on Edison.  Here is the list of packages there.  You can use `conda install --file <file>` to install from this list if you save it to a file.

```
# This file may be used to create an environment using:
# $ conda create --name <env> --file <this file>
# platform: linux-64
backports=1.0=py27_0
backports.shutil_get_terminal_size=1.0.0=py27_1
backports_abc=0.5=py27_0
basemap=1.0.8.dev0=np112py27_3
beautifulsoup4=4.5.3=py27_0
blas=1.1=openblas
bleach=1.5.0=py27_0
bokeh=0.12.4=py27_0
boto3=1.4.4=py27_0
botocore=1.5.26=py27_0
bottleneck=1.2.0=np112py27_0
ca-certificates=2017.1.23=0
cairo=1.14.6=4
cartopy=0.15.1=np112py27_1
certifi=2017.1.23=py27_0
cffi=1.7.0=py27_0
chardet=2.3.0=py27_0
chest=0.2.3=py27_0
click=6.7=py27_0
cloudpickle=0.2.2=py27_2
conda=4.2.13=py27_0
conda-build=2.1.8=py27_0
conda-env=2.6.0=0
conda-verify=2.0.0=py27_0
configparser=3.5.0=py27_0
contextlib2=0.5.4=py27_0
cryptography=1.7.1=py27_0
curl=7.52.1=0
cycler=0.10.0=py27_0
cyordereddict=1.0.0=py27_1
cython=0.25.2=py27_1
dask=0.14.1=py27_0
dbus=1.10.10=3
decorator=4.0.11=py27_0
descartes=1.1.0=py27_0
distributed=1.16.1=py27_0
docutils=0.13.1=py27_0
entrypoints=0.2.2=py27_1
enum34=1.1.6=py27_1
esmf=7.0.0=5
expat=2.1.0=2
fastcache=1.0.2=py27_1
fasteners=0.14.1=py27_2
filelock=2.0.6=py27_0
flake8=3.3.0=py27_0
fontconfig=2.12.1=4
freeglut=3.0.0=4
freetype=2.7=1
freexl=1.0.2=1
funcsigs=1.0.2=py27_0
functools32=3.2.3.2=py27_1
future=0.16.0=py27_0
futures=3.0.5=py27_0
g2clib=1.6.0=2
gdal=2.1.3=np112py27_2
geos=3.5.1=1
get_terminal_size=1.0.0=py27_0
gettext=0.19.7=1
giflib=5.1.4=0
glib=2.51.4=0
gmp=6.1.2=0
graphite2=1.3.9=0
gsl=2.2.1=1
gst-plugins-base=1.8.0=0
gstreamer=1.8.0=1
h5netcdf=0.3.1=py27_0
h5py=2.6.0=np112py27_7
harfbuzz=1.4.3=0
hdf4=4.2.12=0
hdf5=1.8.17=10
hdfeos2=2.19.1=7
hdfeos5=5.1.15=6
heapdict=1.0.0=py27_0
html5lib=0.999=py27_0
icu=58.1=1
idna=2.1=py27_0
ipaddress=1.0.16=py27_0
ipykernel=4.5.2=py27_0
ipython=5.3.0=py27_0
ipython-notebook=4.0.4=py27_0
ipython_genutils=0.2.0=py27_0
jasper=1.900.1=4
jbig=2.1=0
jinja2=2.9.5=py27_0
jmespath=0.9.2=py27_0
jpeg=9b=0
json-c=0.12=0
jsonschema=2.5.1=py27_0
jupyter_client=5.0.0=py27_0
jupyter_core=4.3.0=py27_0
kealib=1.4.6=3
krb5=1.14.2=0
libdap4=3.18.3=2
libffi=3.2.1=3
libgcc=5.2.0=0
libgdal=2.0.0=1
libgfortran=3.0.0=1
libiconv=1.14=4
libnetcdf=4.4.1.1=2
libpng=1.6.28=0
libpq=9.5.4=3
libsodium=1.0.10=0
libspatialite=4.3.0a=14
libtiff=4.0.6=7
libxcb=1.12=1
libxml2=2.9.4=4
libxslt=1.1.29=3
llvmlite=0.16.0=py27_0
locket=0.2.0=py27_1
lxml=3.7.3=py27_0
markupsafe=0.23=py27_1
matplotlib=2.0.0=np112py27_3
mccabe=0.6.1=py27_0
mistune=0.7.4=py27_0
mkl=2017.0.1=0
mock=2.0.0=py27_0
monotonic=1.2=py27_1
mpmath=0.19=py27_2
msgpack-python=0.4.8=py27_0
nbconvert=5.1.1=py27_1
nbformat=4.2.0=py27_0
nco=4.6.5=0
ncurses=5.9=10
netcdf-fortran=4.4.4=2
netcdf4=1.2.7=np112py27_0
nose=1.3.7=py27_2
notebook=4.4.1=py27_0
numexpr=2.6.2=np112py27_0
numpy=1.12.1=py27_blas_openblas_200
olefile=0.44=py27_0
openblas=0.2.19=1
openjpeg=2.1.2=2
openssl=1.0.2h=3
owslib=0.14.0=py27_0
pandas=0.19.2=np112py27_1
pandoc=1.19.2=0
pandocfilters=1.4.1=py27_0
pango=1.40.4=0
partd=0.3.7=py27_0
patchelf=0.9=1
path.py=8.2.1=py27_0
pathlib2=2.2.1=py27_0
patsy=0.4.1=py27_0
pbr=2.0.0=py27_0
pcre=8.39=0
pexpect=4.2.1=py27_0
pickleshare=0.7.3=py27_0
pillow=4.0.0=py27_2
pip=9.0.1=py27_0
pixman=0.34.0=0
pkginfo=1.2.1=py27_0
postgresql=9.5.4=2
progressbar=2.3=py27_0
proj4=4.9.3=3
prompt_toolkit=1.0.14=py27_0
psutil=5.2.1=py27_0
ptyprocess=0.5.1=py27_0
pyasn1=0.2.3=py27_0
pycairo=1.10.0=py27_0
pycodestyle=2.3.1=py27_0
pycosat=0.6.1=py27_0
pycparser=2.17=py27_0
pycrypto=2.6.1=py27_0
pyepsg=0.3.1=py27_0
pyevtk=1.0.0=py27_0
pyflakes=1.5.0=py27_0
pygments=2.2.0=py27_0
pynio=1.5.0_beta20160623=np112py27_14
pyopenssl=16.2.0=py27_0
pyparsing=2.2.0=py27_0
pyproj=1.9.5.1=py27_0
pyqt=5.6.0=py27_0
pyshp=1.2.10=py27_0
python=2.7.13=0
python-dateutil=2.6.0=py27_0
pytz=2016.10=py27_0
pyyaml=3.12=py27_1
pyzmq=16.0.2=py27_1
qt=5.6.2=1
readline=6.2=0
requests=2.13.0=py27_0
ruamel_yaml=0.11.14=py27_0
s3fs=0.0.9=py27_0
s3transfer=0.1.10=py27_1
scandir=1.5=py27_1
scipy=0.19.0=np112py27_blas_openblas_200
seaborn=0.7.1=py27_0
setuptools=33.1.1=py27_0
shapely=1.5.17=np112py27_2
simplegeneric=0.8.1=py27_0
singledispatch=3.4.0.3=py27_0
sip=4.18=py27_1
six=1.10.0=py27_1
sortedcontainers=1.5.3=py27_0
sqlite=3.13.0=1
ssl_match_hostname=3.5.0.1=py27_1
statsmodels=0.8.0=np112py27_0
subprocess32=3.2.7=py27_0
sympy=1.0=py27_0
tblib=1.3.1=py27_0
terminado=0.6=py27_0
testpath=0.3=py27_0
tk=8.5.19=1
toolz=0.8.2=py27_0
tornado=4.4.2=py27_0
traitlets=4.3.2=py27_0
udunits2=2.2.23=0
util-linux=2.21=0
vtk=7.0.0=py27_1
wcwidth=0.1.7=py27_0
webencodings=0.5=py27_0
wheel=0.29.0=py27_0
xarray=0.9.2=py27_0
xerces-c=3.1.4=3
xorg-libxau=1.0.8=3
xorg-libxdmcp=1.1.2=3
xz=5.2.2=0
yaml=0.1.6=0
zeromq=4.2.1=1
zict=0.1.1=py27_0
zlib=1.2.11=0
```

**Comment by xylar** [comment]
Once testing is done and ACME analysis has been updated, we can do the same for the shared install on each machine (`conda install --file <file>`) using the list above.  Then, maybe `conda update --all` for good measure.

**Comment by kevans32** [comment]
Thanks Xylar!

**Comment by xylar** [comment]
@milenaveneziani, I did beta0 and beta1 tests on Edison, which produced the expected HTML and all links seem to work.  (I didn't try every single plot but a random sampling from each task were there.)

I will test again on LANL later tonight.  I think the rest of the testing is now in your hands.  Let me know how it goes, in particular if you run into memory problems.

**Comment by pwolfram** [comment]
Following discussions with @milenaveneziani I'm going to update conda on rhea because there are no production runs currently being performed on rhea or titan.

**Comment by milenaveneziani** [comment]
I am testing on rhea and aims, and moving discussions to ACME PreAndPostProcessingScripts new PR.

**Comment by xylar** [comment]
@milenaveneziani (and @pwolfram if you have an opinion), could you update the checklist above to include whichever PRs you think are necessary before we finalize v0.2?  I'm assuming some (or all) of #175, #177 and #178 will be included.  #178 is not yet merge-ready.  What about #175?

**Comment by milenaveneziani** [comment]
I talked some of this with Phillip as well. We definitely want #177, and @pwolfram was going to check whether we could include #175 since the 2 should be independent. I have #178 on my roadmap with a deadline of Apr 30, so there is no rush for that one.

**Comment by pwolfram** [comment]
I need to do some cleanup on #175.  I'm inclined to include it because it fixes preexisting deficiencies in chunking but am open to alternative suggestions.

**Comment by xylar** [comment]
My understanding is that @pwolfram will be testing and merging each of #179, #169 and #180, at which point we will cut off anymore additions to v0.2 unless further urgent bugs are uncovered in our subsequent testing.

**Comment by pwolfram** [comment]
@xylar and @milenaveneziani, how comprehensive of testing do we need?  Some of these are fairly minor or bug fixes so I'm operating under the philosophy what is important here is a spot-check and them merge for more comprehensive testing in preparation for v0.2-- is this reasonable?  I just want to be on the same page.

**Comment by milenaveneziani** [comment]
@xylar, @pwolfram: I was trying to catch up with emails and github notifications, and then I noticed this. I was under the impression that we already missed yesterday as the day to draw a line and call it v0.2, so let's make Today the day. I don't think we need #169 and #180 right now, but we really need to do more thorough testing within A-Prime, starting on edison asap. Could we agree on this? thanks. (and sorry for slightly frustrated tone)

**Comment by milenaveneziani** [comment]
great that you found the bug in #179 btw! interesting that it didn't show up in testing..

**Comment by xylar** [comment]
@milenaveneziani, I think we both tested on cases where the cache files already existed and thus the code for computing them was bypassed.  That was unfortunate but not that big a deal in the end.

**Comment by xylar** [comment]
@milenaveneziani, At this point, #180 contains bug fixes that we will, indeed, want in v0.2.  But I agree that @pwolfram and I are getting a bit side tracked with fancy extras like deleting corrupted cache files.

**Comment by pwolfram** [comment]
Agreed.  I've merged #180 now.

**Comment by xylar** [comment]
Moving to https://github.com/ACME-Climate/PreAndPostProcessingScripts/pull/29 and closing this issue.
